### PR TITLE
UTF8-encode db paths when matching to fsystem paths

### PR DESCRIPTION
**Pull request by jrwdunham**
Allows transfers containing files with non-ASCII characters in their names.

### CODE REVIEW

File: src/MCPServer/lib/unit.py [hunk 1]
```diff
@@ -60,12 +60,20 @@ def reloadFileList(self):
...
                 files = File.objects.filter(sip_id=self.UUID)
             for f in files:
-                if f.currentlocation in self.fileList:
-                    self.fileList[f.currentlocation].UUID = f.uuid
-                    self.fileList[f.currentlocation].fileGrpUse = f.filegrpuse
+                try:
+                    file_unit = self.fileList.get(
+                        f.currentlocation,
+                        self.fileList.get(f.currentlocation.encode('utf8')))
+                except UnicodeError:
```
Comment (sevein) [inline src/MCPServer/lib/unit.py]: Is it worth also including `KeyError` here?

Comment (jrwdunham) [reply src/MCPServer/lib/unit.py]: I don't think it's possible for `KeyError` to be triggered here. `dict.get(key)` returns `None` if `key` is not in `dict` and `dict.get(key, dflt)` returns `dflt`.

Comment (sevein) [reply src/MCPServer/lib/unit.py]: Ahhh you're right! I keep making that mistake.

File: src/MCPServer/lib/unit.py [hunk 2]
```diff
@@ -60,12 +60,20 @@ def reloadFileList(self):
             else:
                 files = File.objects.filter(sip_id=self.UUID)
             for f in files:
-                if f.currentlocation in self.fileList:
-                    self.fileList[f.currentlocation].UUID = f.uuid
-                    self.fileList[f.currentlocation].fileGrpUse = f.filegrpuse
+                try:
+                    file_unit = self.fileList.get(
+                        f.currentlocation,
+                        self.fileList.get(f.currentlocation.encode('utf8')))
```
Comment (Hwesta) [inline src/MCPServer/lib/unit.py]: Should this use the `archivematicaFunctions.unicodeToStr` ?

**Review by Hwesta — COMMENTED** [review]
This is fairly similar to [this commit](https://github.com/artefactual/archivematica/pull/500/commits/47eeaafba5963bf50a3cc3efeab86b45024e0173). Does that seem like a simpler approach?

**Comment by jrwdunham** [comment]
@Hwesta yes, PR 500 fixes this issue too. I think we want this fix on 1.6 but that PR is deferred to 1.7. What do you suggest? Replace this PR's commit with 47eeaafba5963bf50a3cc3efeab86b45024e0173?

**Comment by Hwesta** [comment]
#500 was a collection of fixes that didn't have bugs, and I made the PR so the potentially useful fixes wouldn't get lost. Since both commits fix the issue, pick whichever one you think is better.  :+1: for this one.

**Comment by jrwdunham** [comment]
Cherry-picked to qa/1.6.x as 358ecc84a2fb31f49b8e0db6c519317eae1785ca

**Pull request merged**